packages/cascara/src/system-components/Boundaries/Boundaries.js

```diff
@@ -1,19 +1,53 @@
-import React from 'react';
+import React, { Suspense } from 'react';
 import pt from 'prop-types';
 import { ErrorBoundary } from 'react-error-boundary';
-import ErrorFallback from './ErrorFallback';
+import { isValidElementType } from 'react-is';
+import { ErrorFallback, SuspenseFallback } from './fallbacks';
+
+// This will need to become a custom prop type we can use elsewhere
+const isReactComponent = (props, propName, componentName) => {
+  if (props[propName] && !isValidElementType(props[propName])) {
+    return new Error(
+      `Invalid prop '${propName}' supplied to '${componentName}': the prop is not a valid React component`
+    );
+  } else {
+    return null;
+  }
+};
 
 const propTypes = {
-  ErrorFallbackComponent: pt.element,
+  ErrorComponent: isReactComponent,
+  SuspenseComponent: isReactComponent,
   children: pt.oneOfType([pt.arrayOf(pt.node), pt.node]),
 };
 
-const Boundaries = ({ ErrorFallbackComponent = ErrorFallback, children }) => {
-  return (
-    <ErrorBoundary FallbackComponent={ErrorFallbackComponent}>
-      {children}
-    </ErrorBoundary>
+const Boundaries = ({
+  ErrorComponent = ErrorFallback,
+  SuspenseComponent = SuspenseFallback,
+  children,
+}) => {
+  // Currently ReactDOMServer does not support Suspense, but we are using it here.
+  // This is a temporary check to make sure that we check to see if we are in a browser,
+  // and if we are, then we are safe to render the boundaries. If we are not, we are on
+  // a server (like Vercel) we an render nothing because the app will render as normal.
+  // once it loads in the client browser and the app hydrates.
+  const isDOM = Boolean(
+    typeof window !== 'undefined' &&
+      window.document &&
+      window.document.createElement
   );
```

Comment:
Just a heads up here that we need to do this to validate if we are running in the DOM or if we are on a server. Our own apps do not need this... but our docs do. This check makes sure we do not try to render the Suspense boundary if we are on server and only try to render if we are in the browser. This is okay since we are not using SSR in our apps.